### DISCUSSION TITLE
Status test runs from 16.08

### DISCUSSION THREAD

**Discussion by maritsandstad** [NorESMhub/CTSM#76] [General]
So, I guess the teams chat is not really visible to most people after, so I thought we could discuss the test runs after today's meeting here

**Comment by maritsandstad** [comment]
FYI I tested the setup from this morning without the accelerated spinup and that built and ran got to the restart going. Haven't really checked the output yet...
@mvdebolskiy , @rosiealice @monsieuralok @kjetilaas 

We tried this thing, @mvertens : ./create_newcase --case test_noresmfates_config --compset 1850_DATM%GSWP3v1_CLM51%FATES_SICE_SOCN_MOSART_SGLC_SWAV  --res ne30pg3_tn14 --run-unsupported --project nn9188k --queue normal --pecount L

**Reply by maritsandstad** [reply]
Looking at the output the restart failed, so I'll look into that...

**Reply by maritsandstad** [reply]
For the direct restart it seemed to have a memory issue, but when I tried to restart manually from what I had, it choked on a missing/empty rpointer file for mosart restart restartfile... There is also no mostart restart file from the first run, which maybe there should be?

**Reply by kjetilaas** [reply]
Thanks for testing this, @maritsandstad. 
I guess this makes sense, since there was a MOSART warning. And we turned those off without fixing it. 
But does it make sense to test without mosart for now?

**Reply by maritsandstad** [reply]
For mine I did not turn off the MOSART warnings because it wasn't the accelerated spinup... I am now testing with the accelerated spinup and I don't get I kept the MOSART warnings on and it still built... So I don't know...

**Reply by rosiealice** [reply]
Thanks Marit!

**Reply by maritsandstad** [reply]
I am happy to do a test without MOSART if this crashes the same way, though. 

For the accelerated spinup I seem to get the same results, out of memory issue for the automatic restart. When I try to resubmit it manually I get the lacking mosart restart file issue...

I'll try with BGC in place of FATES now to see how that goes

**Reply by kjetilaas** [reply]
I guess one more thing we could test is to see if we can start NorESM with the restart files we have, even if there is no mosart restart. 
Since what we want now is initial files for CTSM and FATES, maybe we don't need to sort out the problem with mosart first.

**Reply by mvdebolskiy** [reply]
The reason Rosie's stuff did not work was that she had different externals than the recent tag. I am fixing resubmit and PE layout right now. will give you a new tag soon.

**Reply by maritsandstad** [reply]
Will the new pe_layout fix the memory problem for the immediate restart do you think @mvdebolskiy . I got that even running with BGC and even if I turned MOSART off.

**Reply by maritsandstad** [reply]
But when I turned MOSART off, the manual resubmit restart for BGC, testing the same for the accelerated spinup... But it would be nice if automatic restarts also worked...

**Reply by maritsandstad** [reply]
With MOSART off for accelerated spinup with FATES the automatic restart dies with the memory issue (like they all do), but the manual restart now chokes on a FATES related issue. Seems it's not happy about not finding GPP in the restart file (on inspection the BGC-run restart file for CLM has GPP as a variable (and a bunch of others) that the FATES restart file doesn't seem to have...) @rosiealice maybe you know why and how this can be solved?

**Reply by rosiealice** [reply]
Is it GPP or FATES_GPP?  Steanfw either way as fates definitely normally restarts. I will have a look to see if my runs from the other day have this in their restarts.

**Reply by maritsandstad** [reply]
I think the complaint is about GPP, there is FATES_GPP in the restart file, so is it possible that somewhere the setup doesn't know to read this rather than GPP from the restart file?

**Reply by mvdebolskiy** [reply]
@maritsandstad  accelerated spinup sets history fields in the namelist.  You need to override the history output for FATES.

**Comment by maritsandstad** [comment]
(For reference, the modelgrid_aliases_nuopc.xml in ccs_config had to be change to allow SOCN for the grid setup)

**Comment by mvdebolskiy** [comment]
>I guess one more thing we could test is to see if we can start NorESM with the restart files we have, even if there is no mosart restart.
Since what we want now is initial files for CTSM and FATES, maybe we don't need to sort out the problem with mosart first.

You will get mosart spinup after your AD spinup if you just start the model with finidat='clm2.r.spinup_ad.nc`.

**Comment by rosiealice** [comment]
Are you doing this with accelerated spin up? I don’t think we -need- to do this for this proof of concept run (except if the concept we are proving is the acceleration, which it’s not really)

-------------------------
Dr Rosie A. Fisher

Senior Researcher
CICERO Center for International Climate Research
Oslo, Norway
https://cicero.oslo.no<https://cicero.oslo.no/>
n.b. I am on partial medical leave at present. Thanks for your patience!



________________________________

**Reply by maritsandstad** [reply]
So far I don't really think the accelerated spinup is the main stumbling block... I'll try and push a bit further over the Weekend. Hopefully I can get it to get a viable restart...

**Reply by maritsandstad** [reply]
Without accelerated spinup and MOSART off, the manual restart worked just fine.

**Reply by rosiealice** [reply]
Oh great. Thanks Marit!

**Comment by rosiealice** [comment]
So the sparse grid case I previously set up -is- restarting. (with the restart happening at the end of the year). 

I moved the output to here:
/cluster/projects/nn9188k/rosief/noresm/noresm_ctsm_6_CRU_FATES_NOCOMP_LHCbetzy_test

and the case is here:
/cluster/projects/nn9188k/rosief/noresm/cases/noresm_ctsm_6_CRU_FATES_NOCOMP_LHCbetzy_test

Getting on a plane to Edinburgh now (yesterday did not in fact go according to plan) but will take a closer look and try some more experiments later...

**Comment by mvdebolskiy** [comment]
I have updated ccs_config and config_pes on NorESMhub and made a new tag fates_spinup.

I have  a spinup running with the following script:
```bash
#!/bin/bash

git clone https://github.com/NorESMhub/CTSM.git ctsm_nor

cd ctsm_nor

git checkout fates_spinup

./manage_externals/checkout_externals

set -e

# Created 2024-08-17 21:19:51

CASEDIR="/cluster/projects/nn2806k/mdeb/fates2.5_tests/1850_fates_spinup"

./cime/scripts/create_newcase --case "${CASEDIR}" \
    --compset 1850_DATM%GSWP3v1_CLM51%FATES_SICE_SOCN_MOSART_SGLC_SWAV --res ne30pg3_tn14 \
    --run-unsupported --project nn1004k --queue normal --driver nuopc --pecount L --walltime 26:00:00

cd "${CASEDIR}"

./xmlchange CLM_FORCE_COLDSTART=on,STOP_N=20,STOP_OPTION=nyears,REST_OPTION=nyears,REST_N=10,RESUBMIT=22,MOSART_IGNORE_WARNINGS=TRUE

./case.setup

echo "use_fates_nocomp = .true." >> user_nl_clm

echo "hist_mfilt        = 12 \
hist_nhtfrq       = 0 \
hist_empty_htapes = .true." >> use_nl_clm

./case.build


./case.submit
```

**Reply by rosiealice** [reply]
Great. That looks good to me in terms of settings. If this works I will find my parameter file and code version and try that as well :)

**Comment by maritsandstad** [comment]
Not sure if this is the right place to post this update. But xesmf worked like a dream in regridding the spectral element output, and I can make plots just fine or make lat-lon based netcdf-files for analysis if that's a preference. Happy to share the code if you want but might want to clean up a little bit first. otherwise I used this excellent guide to just do monkey-see monkey-do to get it going: https://ncar.github.io/esds/posts/2022/cam-se-regridding/

**Reply by maritsandstad** [reply]
![lai_output_1850_fates_spinup_80_bias_modis_map png](https://github.com/user-attachments/assets/5fdb89dd-593e-4160-9ea9-a9c5a67835d8)
![vegc_output_1850_fates_spinup_80_absolute_map png](https://github.com/user-attachments/assets/3268c0bb-0a3d-482a-8ad4-5a95407eda8e)
![vegc_output_1850_fates_spinup_160_absolute_map png](https://github.com/user-attachments/assets/0dd21920-10fb-4b86-bbe6-446c76fd083b)
![gpp_output_1850_fates_spinup_80_absolute_map png](https://github.com/user-attachments/assets/dfffc25c-6070-44f4-b9d0-7b766cdfd21c)
![gpp_output_1850_fates_spinup_80_bias_fluxcom_map png](https://github.com/user-attachments/assets/1255ac14-7a29-44b2-ab27-057ca7955213)
![gpp_output_1850_fates_spinup_160_absolute_map png](https://github.com/user-attachments/assets/df77ede3-35f2-4905-9be3-520bb8ddb480)
![gpp_output_1850_fates_spinup_160_bias_fluxcom_map png](https://github.com/user-attachments/assets/0b7e49bf-ede4-44f7-a293-804a091a719e)
![lai_output_1850_fates_spinup_80_absolute_map png](https://github.com/user-attachments/assets/a00599b2-bea1-4582-92cc-a2649f49e745)
![lai_output_1850_fates_spinup_160_absolute_map png](https://github.com/user-attachments/assets/ee1596b1-d75d-4575-9f63-ffc33560119a)
![lai_output_1850_fates_spinup_160_bias_modis_map png](https://github.com/user-attachments/assets/09246eca-9090-4af5-af3b-65e48c01648f)

**Reply by maritsandstad** [reply]
Plots from your spinuprun @mvdebolskiy using said remapping

**Comment by mvertens** [comment]
@maritsandstad - that is fantastic news. How are you handling the masking? I assume that your destination mesh will have no mesh and you simply regrid from a mask grid to an unmask grid?

**Reply by maritsandstad** [reply]
I used a mapfile from clm for the model and grid resolution, I think that took care of the masking (or not) for me. This one (/cluster/shared/noresm/inputdata/lnd/clm2/mappingdata/maps/ne30pg3/map_ne30pg3_to_0.5x0.5_nomask_aave_da_c180515.nc ). It might very well be that I didn't do it the right way, but the results looked reasonable to Kjetil...

**Comment by maritsandstad** [comment]
@mvertens I ran the SP-mode spinup over night. I ran 80 years and that finished in about 12 hours. Output files are here:
/cluster/work/users/masan/archive/1850_fates_spinup/ . Let me know if you want longer/something is wrong or you want details, but I basically just used @mvdebolskiy's script from further up and changed to SP-mode as we discussed.

**Reply by rosiealice** [reply]
Just fwiw, the calibrated parameter file is

https://github.com/rosiealice/fates-global-cal/blob/21b2fe35b8175f7a07056d839b6393b1290046dd/rosie_ppe/gaussian_emulator/parameter_outputs/manual_nocomp_calibration__16july.nc

**Comment by mvertens** [comment]
@maritsandstad - thanks for this. What should finidat be from your spinup run?
@rosiealice - what namelist setting should point to your parameter file (which I will download). Also - for now we are simply going to run in SP mode until we resolve the issue with having the land send back deposition velocities for FATES nocomp.

**Reply by kjetilaas** [reply]
@maritsandstad is traveling today, but as I understand, the finidat should be this file (last restart file from the spinup):
/cluster/work/users/masan/archive/1850_fates_spinup/1850_fates_spinup.clm2.r.0081-01-01-00000.nc

**Comment by mvertens** [comment]
@kjetilaas @maritsandstad @mvdebolskiy - unfortunately I have discovered both a memory leak and restart problems with the non-FATES run I tried to do this weekend. The model crashed at 11 years into the run and when I restarted back a few months - the results were not BFB. However, 11 day restart tests are passing. So right now I have to resolve this before I start the FATES run.
